<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MolcalcWidgetBase</class>
 <widget class="QWidget" name="MolcalcWidgetBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>694</width>
    <height>540</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="calcTab">
      <attribute name="title">
       <string>Calculator</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>16</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout">
         <property name="spacing">
          <number>6</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLineEdit" name="formulaEdit">
           <property name="toolTip">
            <string>Enter molecular formula here</string>
           </property>
           <property name="whatsThis">
            <string>This box can be used to enter the formula for the molecule.
e.g. 1.&gt; CaCO3
e.g. 2.&gt; MeOH
e.g. 3.&gt; #EDTA#
</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="calcButton">
           <property name="toolTip">
            <string>Click on this button to calculate.</string>
           </property>
           <property name="whatsThis">
            <string>Click on this button to calculate.</string>
           </property>
           <property name="text">
            <string>Calc</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="Line" name="line1">
         <property name="frameShape">
          <enum>QFrame::HLine</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="resultComposition">
         <property name="font">
          <font>
           <family>Sans Serif</family>
           <pointsize>12</pointsize>
           <weight>50</weight>
           <italic>false</italic>
           <bold>false</bold>
           <underline>false</underline>
           <strikeout>false</strikeout>
          </font>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="resultMass">
           <property name="text">
            <string/>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="resultValue">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="resultLabel">
         <property name="text">
          <string/>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::MinimumExpanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>41</width>
           <height>131</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QGroupBox" name="details">
         <property name="title">
          <string>Details</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Composition</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTableWidget" name="table">
              <property name="minimumSize">
               <size>
                <width>520</width>
                <height>0</height>
               </size>
              </property>
              <property name="toolTip">
               <string>Elemental composition</string>
              </property>
              <property name="whatsThis">
               <string>This table previews the elemental composition of the molecule, i.e. the percentage of each element in the molecule.</string>
              </property>
              <property name="showGrid">
               <bool>false</bool>
              </property>
              <property name="rowCount">
               <number>1</number>
              </property>
              <property name="columnCount">
               <number>5</number>
              </property>
              <attribute name="horizontalHeaderHighlightSections">
               <bool>true</bool>
              </attribute>
              <row/>
              <column>
               <property name="text">
                <string>Element</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Atoms</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Atomic mass</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Total mass</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Percentage</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Aliases used</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QTableWidget" name="alias_list">
              <property name="toolTip">
               <string>Aliases used in the formula</string>
              </property>
              <property name="whatsThis">
               <string>This displays all the aliases used in the molecule.
For instance, in the molecule MeOH, the alias used is Me = CH3.</string>
              </property>
              <property name="rowCount">
               <number>1</number>
              </property>
              <property name="columnCount">
               <number>1</number>
              </property>
              <row/>
              <column>
               <property name="text">
                <string>aliases</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="aliasTab">
      <attribute name="title">
       <string>Aliases</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QGroupBox" name="aliasBox">
         <property name="title">
          <string>Define alias</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Short-form</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="shortForm">
              <property name="toolTip">
               <string>Short form of the alias</string>
              </property>
              <property name="whatsThis">
               <string>You can specify the short form of the alias used here, 
for instance, Me.</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Full-form</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="fullForm">
              <property name="whatsThis">
               <string>You can enter the full form of the alias in this box, 
for instance, CH3.</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="alias">
              <property name="toolTip">
               <string>Click to add the alias</string>
              </property>
              <property name="whatsThis">
               <string>Click to add the alias if it is valid.</string>
              </property>
              <property name="text">
               <string>Add Alias</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="aliasMessage">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frame">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>300</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Pre-defined aliases</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>User-defined aliases</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QTableWidget" name="pre_defined">
            <property name="toolTip">
             <string>List of pre-defined aliases.</string>
            </property>
            <property name="whatsThis">
             <string>This table displays all the pre-defined aliases.</string>
            </property>
            <property name="rowCount">
             <number>15</number>
            </property>
            <property name="columnCount">
             <number>1</number>
            </property>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <column/>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QTableWidget" name="user_defined">
            <property name="toolTip">
             <string>List of user-defined aliases.</string>
            </property>
            <property name="whatsThis">
             <string>This is the list of aliases that you or some other user has added.</string>
            </property>
            <property name="rowCount">
             <number>15</number>
            </property>
            <property name="columnCount">
             <number>1</number>
            </property>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <row/>
            <column/>
           </widget>
          </item>
          <item row="1" column="1">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>397</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
